### PR TITLE
Add ELK stack and rework build scripts

### PR DESCRIPTION
**Pull request by andybotting**
Make the following changes:
- Use /tmp when installing AWS CFN tools and clean up afterwards
- Add ELK stack feature script and separate Packer config for it
- Rename ubuntu Packer template to default and rework build scripts to build only it by default, but also allow override for building Packer templates
- Split out the pre-cache lines from the Ubuntu Trusty script into its own to be called optionally in Packer templates

### CODE REVIEW

File: packer/resources/features/elk-stack/elasticsearch.yml [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+#
+# For more information, see
+# <http://elasticsearch.org/guide/en/elasticsearch/reference/current/modules-discovery-ec2.html>
+#
+# See <http://elasticsearch.org/tutorials/elasticsearch-on-ec2/>
+# for a step-by-step tutorial.
+
+discovery.type: ec2
+discovery.ec2.tag.Stack: "@@STACK"
+discovery.ec2.tag.App: "@@APP"
```
Comment (sihil) [inline packer/resources/features/elk-stack/elasticsearch.yml:346]: The lack of matching on the Stage tag is presumably merging the PROD and CODE clusters?

File: packer/resources/features/elk-stack/upstart-logcabin.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+stop on shutdown
+
+setuid logcabin
+setgid logcabin
+
+respawn
+
+chdir /opt/logcabin
+
+exec npm start
```
Comment (sihil) [inline packer/resources/features/elk-stack/upstart-logcabin.conf]: I don't think we are using logcabin any more?

File: packer/default.json [hunk 3]
```diff
@@ -65,6 +65,11 @@
     },
     {
       "type": "shell",
+      "script": "resources/pre-cache.sh",
+      "execute_command": "{{ .Vars }} sudo -E bash -x '{{ .Path }}'"
+    },
```
Comment (sihil) [inline packer/default.json:70]: :+1: for this being extracted out

File: packer/resources/features/elk-stack/install.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+#
+# Install ELK stack
+#
+# This script must be run as root
+set -e
+
+FEATURE_ROOT=/opt/features/elk-stack
+KIBANA_VERSION=4.0.2
```
Comment (sihil) [inline packer/resources/features/elk-stack/install.sh]: 4.1.0 is out!

File: packer/resources/features/elk-stack/elasticsearch.yml (hunk 1 above)
Comment (sihil) [inline packer/resources/features/elk-stack/elasticsearch.yml:346]: OK - worked out it is probably also matching on the cluster name which contains the stage.

**Comment by sihil** [comment]
This is all fine - happy with the structure here. 

Probably worth bumping to Kibana 4.1: at this early stage, I imagine it has a good deal of fixes and improvements.

:+1:

File: packer/resources/features/elk-stack/elasticsearch.yml (hunk 1 above)
Comment (andybotting) [inline packer/resources/features/elk-stack/elasticsearch.yml:346]: This was just pulled across from the old config. As you said, it's fixed with the cluster name - but I think I might add stage in here too

**Pull request merged**